client/lib/track-element-size/index.js

```diff
@@ -0,0 +1,92 @@
+/**
+ * External dependencies
+ */
+import { useRef, useEffect, useState, useCallback } from 'react';
+import { throttle } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import afterLayoutFlush from 'lib/after-layout-flush';
+
+const THROTTLE_RATE = 200;
+
+/**
+ * React hook that subscribes a consumer to changes to the bounding client rect of an element, based
+ * on window resize events.
+ * Does not notify when an element resizes due to reasons other than the window resizing.
+ * Uses throttling on the events, to avoid making changes too often.
```

Comment:
is there an advantage to throttling to a fixed time window vs throttling to something adaptive, like `requestAnimationFrame`?

Reply:
I generally avoid using `rAF` for anything which shouldn't be locked to screen refresh rate (anything that isn't an animation, really). Besides being too frequent in this case (at just 16ms), it can become even faster in the future if browsers ever adopt >60fps refresh rates in JS, as Safari has already done in CSS.

Resizes already generally break the frame budget, so there's no point adding more work to that and attempting to recalculate this on every frame.

Reply:
Thanks for the explanation! Sounds like a good candidate for the development devdocs?

Reply:
This explanation, you mean? Sure! Anywhere in particular you had in mind?